### DISCUSSION TITLE
Validators crashing and not producing new blocks ERROR : "out of gas"

### DISCUSSION THREAD

**Discussion by ex-shubham-koli** [0xPolygon/polygon-edge#792] [Q&A]
Earlier today all my validators crashed and all of them are showing the following logs. I have 8 validator setup on AWS. with 4 Validators created at first and then 4 added later via smart contract. (all BLS supported)

My env
Polygon edge v0.6
Ubuntu 20

> 0|node7    | 2022-10-12T10:04:41.030Z [INFO]  polygon.blockchain: genesis: hash=0x4a5151d9bab06e1da34f05efc5e91c361ce33a58515ea19b21ca7d67c4893c38
0|node7    | out of gas

any idea why am I getting this error? any help is appreciated.

![image](https://user-images.githubusercontent.com/53253101/195315318-f91991fc-4a4a-49a7-a572-b85f29dd7379.png)

![image](https://user-images.githubusercontent.com/53253101/195315217-a2477a8e-5fcc-47b3-8da5-9a3822bd2e35.png)

**Comment by laviniat1996** [comment]
Hi @shubham-edx ,

Did that error happen out of nowhere or after you did a certain action? Was your chain running fine until this error, and since when was it running?

Also, the full logs would be very apreciated. Thank you.

**Reply by ex-shubham-koli** [reply]
Hi @laviniat1996 ,
This error happened out of nowhere. not sure if this is related but I staked X amount of tokens for the new validator. this was at block [185041](https://explorer.edexa.com/block/185041) and then my validators stopped working at block [185099](https://explorer.edexa.com/block/185041). my epoch size is 100 blocks. this blockchain has been running for the last 8 days. longest till now.
link to logs of [one node](https://drive.google.com/file/d/1waoLPNSwWcOqCnej24zfzoahrdToQFvr/view?usp=sharing
)
let me know if you would like me to share logs of all nodes.

**Reply by laviniat1996** [reply]
@shubham-edx How did you stake the tokens (what were the steps)? It seems that the network stopped after this action, when it hit the next epoch.

**Reply by ex-shubham-koli** [reply]
Hi @laviniat1996 ,
nothing irregular. I used provided scripts to make the tx and register the BLS key. is there any way I can start the blockchain again from block 185040?

**Reply by laviniat1996** [reply]
@shubham-edx Thank you for the info. How did you do the update to v0.6.0? Please note that this is a breaking change, we have a guide for updating the nodes here: https://wiki.polygon.technology/docs/edge/validator-hosting#update

**Reply by ex-shubham-koli** [reply]
@laviniat1996 v0.6

**Reply by ex-shubham-koli** [reply]
I started a new network from scratch using v0.6

**Reply by laviniat1996** [reply]
You are running in PoS right?

**Reply by ex-shubham-koli** [reply]
@laviniat1996, Yes I am running POS.
if you could point me to the document or tell me how to start the blockchain from X block? that would be helpful.

**Reply by laviniat1996** [reply]
@shubham-edx Unfortunately, you cannot start from a certain block. It seems that the `out of gas ` error might indicate that some contract call was not executed due to lack of gas. Did you set the `block-gas-limit`?

**Reply by ex-shubham-koli** [reply]
Yes while creating the genesis file I have set --block-gas-limit 1000000000.

**Reply by laviniat1996** [reply]
@shubham-edx Can you please tell me what's your `price-limit`? I'm trying to replicate this on my end.

**Reply by ex-shubham-koli** [reply]
I am not sure if I have setup any price limit variable manually. perhaps you mean gas limit?
this is my genesis file.

> {
"name": "Testnet",
"genesis": {
"nonce": "0x0000000000000000",
"timestamp": "0x0",
"extraData": "0x0000000000000000000000000000000000000000000000000000000000000000f90127f90120f84694d51c5787aef3346f1842b3796e68bf555c66c57db0a060b82fbcf9cbd94a9a6e2a02fd4f9905dab2d6a62f3f72a0953122eb2f385e7bf8f99a6195be97c7fc451e1a5f2199f84694d52ec084e1032fc89b9126d39e3be51c885a8203b0974d2462b5cd9016c31cf80cb4c094c940e51be6fefd1b503e160845d756e26950b58a2e3c1d905ee766098427543546f84694833dd878f3b229b8c4c18d8c5cdc1ac65c35dcfab0912a08ee3d05df723fb3b6f0df41b648212f09dff3213b2d310262fab9342bff3b451cedaceddd0e1f60bd248c04862bf8469492507ab8f1d898225b7f8dda578af7839c09e4e5b099a33abfb39ea49cf2f6bb813d811d841de8436ce5dcf481b6393d384fdde1d000935ea50d542600a5176529068a52ee80c28080",
"gasLimit": "0x3b9aca00",
"difficulty": "0x1",
"mixHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
"coinbase": "0x0000000000000000000000000000000000000000",
"alloc": {
"0x0000000000000000000000000000000000001001": {
"code": "0x6080604052600436106101185760003560e01c80637a6eea37116100a0578063d94c111b11610064578063d94c111b1461040a578063e387a7ed14610433578063e804fbf61461045e578063f90ecacc14610489578063facd743b146104c657610186565b80637a6eea37146103215780637dceceb81461034c578063af6da36e14610389578063c795c077146103b4578063ca1e7819146103df57610186565b8063373d6132116100e7578063373d6132146102595780633a4b66f1146102845780633c561f041461028e57806351a9ab32146102b9578063714ff425146102f657610186565b806302b751991461018b578063065ae171146101c85780632367f6b5146102055780632def66201461024257610186565b366101865761013c3373ffffffffffffffffffffffffffffffffffffffff16610503565b1561017c576040517f08c379a0000000000000000000000000000000000000000000000000000000008152600401610173906117ec565b60405180910390fd5b610184610526565b005b600080fd5b34801561019757600080fd5b506101b260048036038101906101ad91906113e2565b6105fd565b6040516101bf9190611847565b60405180910390f35b3480156101d457600080fd5b506101ef60048036038101906101ea91906113e2565b610615565b6040516101fc919061174f565b60405180910390f35b34801561021157600080fd5b5061022c600480360381019061022791906113e2565b610635565b6040516102399190611847565b60405180910390f35b34801561024e57600080fd5b5061025761067e565b005b34801561026557600080fd5b5061026e610769565b60405161027b9190611847565b60405180910390f35b61028c610773565b005b34801561029a57600080fd5b506102a36107dc565b6040516102b0919061172d565b60405180910390f35b3480156102c557600080fd5b506102e060048036038101906102db91906113e2565b610982565b6040516102ed919061176a565b60405180910390f35b34801561030257600080fd5b5061030b610a22565b6040516103189190611847565b60405180910390f35b34801561032d57600080fd5b50610336610a2c565b604051610343919061182c565b60405180910390f35b34801561035857600080fd5b50610373600480360381019061036e91906113e2565b610a3a565b6040516103809190611847565b60405180910390f35b34801561039557600080fd5b5061039e610a52565b6040516103ab9190611847565b60405180910390f35b3480156103c057600080fd5b506103c9610a58565b6040516103d69190611847565b60405180910390f35b3480156103eb57600080fd5b506103f4610a5e565b604051610401919061170b565b60405180910390f35b34801561041657600080fd5b50610431600480360381019061042c919061140f565b610aec565b005b34801561043f57600080fd5b50610448610b91565b6040516104559190611847565b60405180910390f35b34801561046a57600080fd5b50610473610b97565b6040516104809190611847565b60405180910390f35b34801561049557600080fd5b506104b060048036038101906104ab9190611458565b610ba1565b6040516104bd91906116f0565b60405180910390f35b3480156104d257600080fd5b506104ed60048036038101906104e891906113e2565b610be0565b6040516104fa919061174f565b60405180910390f35b6000808273ffffffffffffffffffffffffffffffffffffffff163b119050919050565b34600460008282546105389190611968565b9250508190555034600260003373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff168152602001908152602001600020600082825461058e9190611968565b9250508190555061059e33610c36565b156105ad576105ac33610cb0565b5b3373ffffffffffffffffffffffffffffffffffffffff167f9e71bc8eea02a63969f509818f2dafb9254532904319f9dbda79b67bd34a5f3d346040516105f39190611847565b60405180910390a2565b60036020528060005260406000206000915090505481565b60016020528060005260406000206000915054906101000a900460ff1681565b6000600260008373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff168152602001908152602001600020549050919050565b61069d3373ffffffffffffffffffffffffffffffffffffffff16610503565b156106dd576040517f08c379a00000000000000000000000000000000000000000000000000000000081526004016106d4906117ec565b60405180910390fd5b6000600260003373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff168152602001908152602001600020541161075f576040517f08c379a00000000000000000000000000000000000000000000000000000000081526004016107569061178c565b60405180910390fd5b610767610dff565b565b6000600454905090565b6107923373ffffffffffffffffffffffffffffffffffffffff16610503565b156107d2576040517f08c379a00000000000000000000000000000000000000000000000000000000081526004016107c9906117ec565b60405180910390fd5b6107da610526565b565b60606000808054905067ffffffffffffffff8111156107fe576107fd611c00565b5b60405190808252806020026020018201604052801561083157816020015b606081526020019060019003908161081c5790505b50905060005b60008054905081101561097a576007600080838154811061085b5761085a611bd1565b5b9060005260206000200160009054906101000a900473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002080546108cb90611a98565b80601f01602080910402602001604051908101604052809291908181526020018280546108f790611a98565b80156109445780601f1061091957610100808354040283529160200191610944565b820191906000526020600020905b81548152906001019060200180831161092757829003601f168201915b505050505082828151811061095c5761095b611bd1565b5b6020026020010181905250808061097290611afb565b915050610837565b508091505090565b600760205280600052604060002060009150905080546109a190611a98565b80601f01602080910402602001604051908101604052809291908181526020018280546109cd90611a98565b8015610a1a5780601f106109ef57610100808354040283529160200191610a1a565b820191906000526020600020905b8154815290600101906020018083116109fd57829003601f168201915b505050505081565b6000600554905090565b69021e19e0c9bab240000081565b60026020528060005260406000206000915090505481565b60065481565b60055481565b60606000805480602002602001604051908101604052809291908181526020018280548015610ae257602002820191906000526020600020905b8160009054906101000a900473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1681526020019060010190808311610a98575b5050505050905090565b80600760003373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1681526020019081526020016000209080519060200190610b3f9291906112a5565b503373ffffffffffffffffffffffffffffffffffffffff167f472da4d064218fa97032725fbcff922201fa643fed0765b5ffe0ceef63d7b3dc82604051610b86919061176a565b60405180910390a250565b60045481565b6000600654905090565b60008181548110610bb157600080fd5b906000526020600020016000915054906101000a900473ffffffffffffffffffffffffffffffffffffffff1681565b6000600160008373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002060009054906101000a900460ff169050919050565b6000610c4182610f51565b158015610ca9575069021e19e0c9bab24000006fffffffffffffffffffffffffffffffff16600260008473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1681526020019081526020016000205410155b9050919050565b60065460008054905010610cf9576040517f08c379a0000000000000000000000000000000000000000000000000000000008152600401610cf0906117ac565b60405180910390fd5b60018060008373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002060006101000a81548160ff021916908315150217905550600080549050600360008373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff168152602001908152602001600020819055506000819080600181540180825580915050600190039060005260206000200160009091909190916101000a81548173ffffffffffffffffffffffffffffffffffffffff021916908373ffffffffffffffffffffffffffffffffffffffff16021790555050565b6000600260003373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1681526020019081526020016000205490506000600260003373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff168152602001908152602001600020819055508060046000828254610e9a91906119be565b92505081905550610eaa33610f51565b15610eb957610eb833610fa7565b5b3373ffffffffffffffffffffffffffffffffffffffff166108fc829081150290604051600060405180830381858888f19350505050158015610eff573d6000803e3d6000fd5b503373ffffffffffffffffffffffffffffffffffffffff167f0f5bb82176feb1b5e747e28471aa92156a04d9f3ab9f45f28e2d704232b93f7582604051610f469190611847565b60405180910390a250565b6000600160008373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002060009054906101000a900460ff169050919050565b60055460008054905011610ff0576040517f08c379a0000000000000000000000000000000000000000000000000000000008152600401610fe79061180c565b60405180910390fd5b600080549050600360008373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1681526020019081526020016000205410611076576040517f08c379a000000000000000000000000000000000000000000000000000000000815260040161106d906117cc565b60405180910390fd5b6000600360008373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff168152602001908152602001600020549050600060016000805490506110ce91906119be565b90508082146111bc5760008082815481106110ec576110eb611bd1565b5b9060005260206000200160009054906101000a900473ffffffffffffffffffffffffffffffffffffffff169050806000848154811061112e5761112d611bd1565b5b9060005260206000200160006101000a81548173ffffffffffffffffffffffffffffffffffffffff021916908373ffffffffffffffffffffffffffffffffffffffff16021790555082600360008373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002081905550505b6000600160008573ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002060006101000a81548160ff0219169083151502179055506000600360008573ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002081905550600080548061126b5761126a611ba2565b5b6001900381819060005260206000200160006101000a81549073ffffffffffffffffffffffffffffffffffffffff02191690559055505050565b8280546112b190611a98565b90600052602060002090601f0160209004810192826112d3576000855561131a565b82601f106112ec57805160ff191683800117855561131a565b8280016001018555821561131a579182015b828111156113195782518255916020019190600101906112fe565b5b509050611327919061132b565b5090565b5b8082111561134457600081600090555060010161132c565b5090565b600061135b61135684611887565b611862565b90508281526020810184848401111561137757611376611c34565b5b611382848285611a56565b509392505050565b60008135905061139981611d6d565b92915050565b600082601f8301126113b4576113b3611c2f565b5b81356113c4848260208601611348565b91505092915050565b6000813590506113dc81611d84565b92915050565b6000602082840312156113f8576113f7611c3e565b5b60006114068482850161138a565b91505092915050565b60006020828403121561142557611424611c3e565b5b600082013567ffffffffffffffff81111561144357611442611c39565b5b61144f8482850161139f565b91505092915050565b60006020828403121561146e5761146d611c3e565b5b600061147c848285016113cd565b91505092915050565b600061149183836114b1565b60208301905092915050565b60006114a983836115b1565b905092915050565b6114ba816119f2565b82525050565b6114c9816119f2565b82525050565b60006114da826118d8565b6114e48185611913565b93506114ef836118b8565b8060005b838110156115205781516115078882611485565b9750611512836118f9565b9250506001810190506114f3565b5085935050505092915050565b6000611538826118e3565b6115428185611924565b935083602082028501611554856118c8565b8060005b858110156115905784840389528151611571858261149d565b945061157c83611906565b925060208a01995050600181019050611558565b50829750879550505050505092915050565b6115ab81611a04565b82525050565b60006115bc826118ee565b6115c68185611935565b93506115d6818560208601611a65565b6115df81611c43565b840191505092915050565b60006115f5826118ee565b6115ff8185611946565b935061160f818560208601611a65565b61161881611c43565b840191505092915050565b6000611630601d83611957565b915061163b82611c54565b602082019050919050565b6000611653602783611957565b915061165e82611c7d565b604082019050919050565b6000611676601283611957565b915061168182611ccc565b602082019050919050565b6000611699601a83611957565b91506116a482611cf5565b602082019050919050565b60006116bc604083611957565b91506116c782611d1e565b604082019050919050565b6116db81611a10565b82525050565b6116ea81611a4c565b82525050565b600060208201905061170560008301846114c0565b92915050565b6000602082019050818103600083015261172581846114cf565b905092915050565b60006020820190508181036000830152611747818461152d565b905092915050565b600060208201905061176460008301846115a2565b92915050565b6000602082019050818103600083015261178481846115ea565b905092915050565b600060208201905081810360008301526117a581611623565b9050919050565b600060208201905081810360008301526117c581611646565b9050919050565b600060208201905081810360008301526117e581611669565b9050919050565b600060208201905081810360008301526118058161168c565b9050919050565b60006020820190508181036000830152611825816116af565b9050919050565b600060208201905061184160008301846116d2565b92915050565b600060208201905061185c60008301846116e1565b92915050565b600061186c61187d565b90506118788282611aca565b919050565b6000604051905090565b600067ffffffffffffffff8211156118a2576118a1611c00565b5b6118ab82611c43565b9050602081019050919050565b6000819050602082019050919050565b6000819050602082019050919050565b600081519050919050565b600081519050919050565b600081519050919050565b6000602082019050919050565b6000602082019050919050565b600082825260208201905092915050565b600082825260208201905092915050565b600082825260208201905092915050565b600082825260208201905092915050565b600082825260208201905092915050565b600061197382611a4c565b915061197e83611a4c565b9250827fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff038211156119b3576119b2611b44565b5b828201905092915050565b60006119c982611a4c565b91506119d483611a4c565b9250828210156119e7576119e6611b44565b5b828203905092915050565b60006119fd82611a2c565b9050919050565b60008115159050919050565b60006fffffffffffffffffffffffffffffffff82169050919050565b600073ffffffffffffffffffffffffffffffffffffffff82169050919050565b6000819050919050565b82818337600083830152505050565b60005b83811015611a83578082015181840152602081019050611a68565b83811115611a92576000848401525b50505050565b60006002820490506001821680611ab057607f821691505b60208210811415611ac457611ac3611b73565b5b50919050565b611ad382611c43565b810181811067ffffffffffffffff82111715611af257611af1611c00565b5b80604052505050565b6000611b0682611a4c565b91507fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff821415611b3957611b38611b44565b5b600182019050919050565b7f4e487b7100000000000000000000000000000000000000000000000000000000600052601160045260246000fd5b7f4e487b7100000000000000000000000000000000000000000000000000000000600052602260045260246000fd5b7f4e487b7100000000000000000000000000000000000000000000000000000000600052603160045260246000fd5b7f4e487b7100000000000000000000000000000000000000000000000000000000600052603260045260246000fd5b7f4e487b7100000000000000000000000000000000000000000000000000000000600052604160045260246000fd5b600080fd5b600080fd5b600080fd5b600080fd5b6000601f19601f8301169050919050565b7f4f6e6c79207374616b65722063616e2063616c6c2066756e6374696f6e000000600082015250565b7f56616c696461746f72207365742068617320726561636865642066756c6c206360008201527f6170616369747900000000000000000000000000000000000000000000000000602082015250565b7f696e646578206f7574206f662072616e67650000000000000000000000000000600082015250565b7f4f6e6c7920454f412063616e2063616c6c2066756e6374696f6e000000000000600082015250565b7f56616c696461746f72732063616e2774206265206c657373207468616e20746860008201527f65206d696e696d756d2072657175697265642076616c696461746f72206e756d602082015250565b611d76816119f2565b8114611d8157600080fd5b50565b611d8d81611a4c565b8114611d9857600080fd5b5056fea26469706673582212202736be5ece8930592087c8493ea688ec518e8e5cb859b3f5f5d50fe7886c4acf64736f6c63430008070033",
"storage": {
"0x0000000000000000000000000000000000000000000000000000000000000000": "0x0000000000000000000000000000000000000000000000000000000000000004",
"0x0000000000000000000000000000000000000000000000000000000000000004": "0x0000000000000000000000000000000000000000000000022b1c8c1227a00000",
"0x0000000000000000000000000000000000000000000000000000000000000005": "0x0000000000000000000000000000000000000000000000000000000000000001",
"0x0000000000000000000000000000000000000000000000000000000000000006": "0x000000000000000000000000000000000000000000000000001ffffffffffffe",
"0x290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563": "0x000000000000000000000000d51c5787aef3346f1842b3796e68bf555c66c57d",
"0x290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e564": "0x000000000000000000000000d52ec084e1032fc89b9126d39e3be51c885a8203",
"0x290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e565": "0x000000000000000000000000833dd878f3b229b8c4c18d8c5cdc1ac65c35dcfa",
"0x290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e566": "0x00000000000000000000000092507ab8f1d898225b7f8dda578af7839c09e4e5",
"0x50ccc901522efa269ceb86e13bddd53bff8291d41dedba2f5ac999f319993aac": "0x0000000000000000000000000000000000000000000000000000000000000001",
"0x610dc3ad652825d03c85872355d0b1737eacad02e96b56c0fab1a62ac171803b": "0x0000000000000000000000000000000000000000000000008ac7230489e80000",
"0x6547087cbbaef2c9709ce5f74e1f1f9e7d5599508dfe0c1f9bcb9a0444428c5b": "0x99a33abfb39ea49cf2f6bb813d811d841de8436ce5dcf481b6393d384fdde1d0",
"0x6547087cbbaef2c9709ce5f74e1f1f9e7d5599508dfe0c1f9bcb9a0444428c5c": "0x00935ea50d542600a5176529068a52ee00000000000000000000000000000000",
"0x6bd64f97512294eece42376a4975f62e310f14ce88585eaa3d0983f0fab5cb3f": "0x0000000000000000000000000000000000000000000000000000000000000000",
"0x77677af7ac253de53c00fa0655463c9471118573ef38c3560adab12ff9b422d3": "0x0000000000000000000000000000000000000000000000000000000000000061",
"0x909792bc309100e7a020a490b9457c77ce147e33e6f46cc9f27d105f2f6775ef": "0x0000000000000000000000000000000000000000000000000000000000000001",
"0x950c796158dee80830b59eec179ef15a407fbacbb6c3945e5c2478947bb70f2e": "0x0000000000000000000000000000000000000000000000000000000000000001",
"0x9d97ada8c20e3616858b61c4e5739847d45ec56b515097398a11ab744acc2656": "0x0000000000000000000000000000000000000000000000000000000000000003",
"0xa2ed53a055c0be4079bbc0b1892ca22764773ebb005f98a31e0441910b69db45": "0x0000000000000000000000000000000000000000000000000000000000000001",
"0xa8cd0c4ff9b7431aff3279c0a5926e46ba09b9409ea094367b6f8debd81f410e": "0x0000000000000000000000000000000000000000000000000000000000000061",
"0xb117bf1112626ebd9f0bf0d16779ffdc61de1484dccb43187bd5440379ed2990": "0x0000000000000000000000000000000000000000000000008ac7230489e80000",
"0xb291bd1f065f601367dd3918a21a67ce3f834d8420f60f68c10b80adc3b4a043": "0x0000000000000000000000000000000000000000000000000000000000000061",
"0xcaf6f7179a38a2acb2ac7405a2bef094a4f90431b9032c4199a294a655e6c6dc": "0x0000000000000000000000000000000000000000000000000000000000000002",
"0xcf73fd7c64f64ac190cd53d484a5cfb1714fb79e0e36591b00b03f4275aabba2": "0x0000000000000000000000000000000000000000000000008ac7230489e80000",
"0xd18907825ac0235ec6bc3dcbb934d6470164e029a74904284a23d1d111afedda": "0xa060b82fbcf9cbd94a9a6e2a02fd4f9905dab2d6a62f3f72a0953122eb2f385e",
"0xd18907825ac0235ec6bc3dcbb934d6470164e029a74904284a23d1d111afeddb": "0x7bf8f99a6195be97c7fc451e1a5f219900000000000000000000000000000000",
"0xdde29a361831b2b14b4ec3d45277b56a5a3b297fdfa47d686535e87a42c02cb5": "0x0000000000000000000000000000000000000000000000000000000000000001",
"0xe0271979868ea4ea2a03969d7d415f165b30f8327bd11db2a229214c01b8b09f": "0x0000000000000000000000000000000000000000000000000000000000000061",
"0xecbe740014af985f3c2a23449e858aa9b9ef0ea794251280778dc2e2fd202a46": "0x0000000000000000000000000000000000000000000000008ac7230489e80000",
"0xf54e14d24600bfb7e32d59db68be11bfa11ab7cbca8286a041f028dbd3f20183": "0x974d2462b5cd9016c31cf80cb4c094c940e51be6fefd1b503e160845d756e269",
"0xf54e14d24600bfb7e32d59db68be11bfa11ab7cbca8286a041f028dbd3f20184": "0x50b58a2e3c1d905ee76609842754354600000000000000000000000000000000",
"0xfc634297989d2c5322da8b09e44b26cd939f258244c2978bddaf3b6c52966f3d": "0x912a08ee3d05df723fb3b6f0df41b648212f09dff3213b2d310262fab9342bff",
"0xfc634297989d2c5322da8b09e44b26cd939f258244c2978bddaf3b6c52966f3e": "0x3b451cedaceddd0e1f60bd248c04862b00000000000000000000000000000000"
},
"balance": "0x22b1c8c1227a00000"
},
"0x42b8d75b02dF61Bb9c11159c4902d1Ea34BaDacd": {
"balance": "0x204fce5e3e25026110000000"
}
},
"number": "0x0",
"gasUsed": "0x70000",
"parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000"
},
"params": {
"forks": {
"homestead": 0,
"byzantium": 0,
"constantinople": 0,
"petersburg": 0,
"istanbul": 0,
"EIP150": 0,
"EIP158": 0,
"EIP155": 0
},
"chainID": 1995,
"engine": {
"ibft": {
"epochSize": 100,
"type": "PoS",
"validator_type": "bls"
}
},
"blockGasTarget": 0
},
"bootnodes": [
"/ip4/<removed>/tcp/1478/p2p/16Uiu2HAmNm9G7pY3aouwTjyHksdwRp5nW3LJWT8mnyhwajtK1Dqf",
"/ip4/<removed>/tcp/2478/p2p/16Uiu2HAmUxdEopRRPV9PGWfq2p89pHXYfSPCxowCcxeDStQxF7EH",
"/ip4/<removed>/tcp/1478/p2p/16Uiu2HAm7Y6vKzK5wVTWhvatu4B2GUqCzFmp4Gvr28KU7G6EDbzA",
"/ip4/<removed>/tcp/2478/p2p/16Uiu2HAmPgQWfZWTtJhziU3WKGsTnRevNWioF4M3yfwAwNhFt9wk",
"/ip4/<removed>/tcp/1478/p2p/16Uiu2HAmMAG4YVt55AKaJedmnnrTvNUnR68jV8VTaj1eU5jFSVmi",
"/ip4/<removed>/tcp/2478/p2p/16Uiu2HAmHQvSGoB2GezZdHkCwyiX6mA53RN8B9SBZtP2CTTcHCVG",
"/ip4/<removed>/tcp/1478/p2p/16Uiu2HAm7D7eB3nTBo5zMdJcbJNYXbrkiSWBX8iDCj7efERNYbjf",
"/ip4/<removed>/tcp/2478/p2p/16Uiu2HAmMBRY64fh2wB5vrw42zeFM5fR5LrfwvRiXbDkHhakZXW1"
]
}

**Reply by laviniat1996** [reply]
@shubham-edx Hi, thank you very much for reporting this. After discussing the issue with the team, we found the root of it. There is a transaction made for fetching the validator set from the contract each epoch. This transaction is not made by the user, it's created locally and executed. 
The gas limit for this transaction is:  
```
// Gas limit used when querying the validator set
queryGasLimit uint64 = 100000
```
**This is too small to execute that transaction when you have too many validators** (by the way, in the logs it shows that you have 10 validators actually `validators=10 committed=7`).

**We are working on a solution for this problem.**

What I will kindly ask you is to open up an issue regarding this, where we will keep you updated.

**Reply by ex-shubham-koli** [reply]
okay, thanks for the help. I have created an issue for this.
